app/decorators/post_search_result_set_decorator.rb

```diff
@@ -1,22 +1,20 @@
 module PostSearchResultSetDecorator
   def canonical_params
-    {query: {keywords: keywords}, page: posts.current_page}
+    {query: {keywords: keywords}, page: paginator.current_page}
   end
 
   def canonical_url
     url_for(canonical_params.merge(only_path: false))
   end
 
   def message
-    if keywords.empty?
-      "検索キーワードを入力してください。"
-    elsif posts.empty?
-      "「#{keywords}」を含む記事は見つかりませんでした。"
-    elsif posts.total_pages > 1
-      "「#{keywords}」を含む記事は#{posts.total_count}件見つかりました。(#{posts.page_entries_info})"
-    else
-      "「#{keywords}」を含む記事は#{posts.total_count}件見つかりました。"
-    end
+    return nil if keywords.empty?
+
+    message = []
+    message << "「#{keywords}」を含む"
+    message << (posts.empty? ?  "記事は見つかりませんでした。" : "記事が#{paginator.total_count}件見つかりました。")
+    message << "(#{page_entries_info})" if paginator.total_pages > 1
+    message.join
```

Comment:
In Ruby 3.0, string literal may be frozen. https://bugs.ruby-lang.org/issues/11473
I want to avoid using the code that may be broken sometime in the future.